app/src/main/java/com/okta/android/samples/browser_sign_in/BrowserSignInActivity.kt

```diff
@@ -0,0 +1,68 @@
+package com.okta.android.samples.browser_sign_in
+
+import android.os.Bundle
+import android.view.View
+import android.widget.TextView
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+
+class BrowserSignInActivity : AppCompatActivity() {
+    private val viewModel by viewModels<BrowserSignInViewModel>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_browser_sign_in)
+
+        val progressBar = findViewById<View>(R.id.progress_bar)
+        val statusTextView = findViewById<TextView>(R.id.status_text_view)
+        val loginButton = findViewById<View>(R.id.login_button)
+        val logoutOfBrowserButton = findViewById<View>(R.id.logout_of_browser_button)
+
+        loginButton.setOnClickListener {
+            viewModel.login(this)
+        }
+        logoutOfBrowserButton.setOnClickListener {
+            viewModel.logoutOfBrowser(this)
+        }
+
+        /**
+         * Observe the [BrowserSignInViewModel]s [BrowserState] for changes.
+         *
+         * Changes are initialized by the button clicks above, and happen asynchronously once the user completes the flows.
+         *
+         * Once the change occurs, show the user the correct UI.
+         */
```

Comment:
Is the reference to other objects necessary for the summary? Perhaps something like:

```suggestion
        /**
         * Update the user interface for changes in the sign-in flow.
         *
         * Use an observer to react to updates in [BrowserState]. Updates are asynchronous and are triggered both by user actions, such as button clicks, and completing the flow.
         */
```

The structure is similar to the sample overview. It starts with a one line abstract and then gives more detail.